src/cloudflare/internal/d1-api.ts

```diff
@@ -266,7 +266,7 @@ class D1PreparedStatement {
     )
   }
 
-  public async all<T = unknown>(): Promise<D1Result<T[]>> {
+  public async all<T = Record<string, unknown>>(): Promise<D1Result<T[]>> {
```

Comment:
So the default return type here is:

```ts
Promise<D1Result<Record<string, unknown>[]>>
```

So the returned object will have a `results` property which is an array of:

```ts
Record<string, unknown>[]
```

So each row of the `results` will be an array of records. Is that what you want?